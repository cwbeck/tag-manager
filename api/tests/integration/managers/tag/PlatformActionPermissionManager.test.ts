import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import PlatformActionPermissionRepo from '../../../../src/mongo/repos/tag/PlatformActionPermissionRepo';
import Mock = jest.Mock;

describe('PlatformActionPermissionManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single platform action permission', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatformRevision();

        const mockSutModel = TestUtils.buildMockPlatformActionPermission();
        const mockSutRepo = {} as PlatformActionPermissionRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatformActionPermission(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    permission
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
