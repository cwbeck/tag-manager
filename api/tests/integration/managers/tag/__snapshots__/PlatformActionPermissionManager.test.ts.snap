// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformActionPermissionManager fetches a single platform action permission 1`] = `
Object {
  "data": Object {
    "getPlatformActionPermission": Object {
      "id": "5edf4c48e9615d9a1c66caef",
      "permission": "LOG_TO_CONSOLE",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
